docs/book/src/plugins/creating-plugins.md

```diff
@@ -64,3 +64,46 @@ Feel free to check the implementation under:
 
 - deploy-image: <https://github.com/kubernetes-sigs/kubebuilder/tree/v3.7.0/pkg/plugins/golang/deploy-image/v1alpha1>
 - grafana: <https://github.com/kubernetes-sigs/kubebuilder/tree/v3.7.0/pkg/plugins/optional/grafana/v1alpha>
+
+## Plugin scaffolding
+
+Your plugin may have the use case to add code on top of what is scaffolded by default with `create api`.
+This is common as you may expect your plugin to:
+
+- Create API
+- Update controller manager logic
+- Generate corresponding manifests
+
+Following approaches may help:
+
+- If your plugin is expected to overwrite existing files with its own scaffold, you can set the `machinary.IfExistsAction` value to be `OverwriteFile`. Thus, when the plugin is called to scaffold, the target file will be overwritten if existing. See [example of deploy-image][example-of-deploy-image-1].
+- If your plugin is expected to modify part of the existing files with its scaffold, you may use functions provided by [sigs.k8s.io/kubebuilder/v3/pkg/plugin/util][kb-util]. See [example of deploy-image][example-of-deploy-image-2].
+
+#### Simplify the calling of multiple plugins
```

Comment:
That is not for simplicity. Give a look at "Composite Pattern"
The idea is be able to re-use the components which provides pros like better maintainability.